<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Mem_Allocator</class>
 <widget class="QMainWindow" name="Mem_Allocator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1600</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mem_Allocator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>1600</width>
     <height>900</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">#centralwidget {
	border-image: url(:/Icons/Milkyway.jpeg)
}

</string>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>50</y>
      <width>501</width>
      <height>801</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="styleSheet">
        <string notr="true">#groupBox {
	background-color: rgb(200, 200, 200);
}</string>
       </property>
       <property name="title">
        <string/>
       </property>
       <widget class="QFrame" name="frame">
        <property name="geometry">
         <rect>
          <x>19</x>
          <y>69</y>
          <width>151</width>
          <height>641</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <widget class="QRadioButton" name="Initialize">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>151</width>
           <height>51</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Initialize</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="Allocate">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>220</y>
           <width>151</width>
           <height>51</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Allocate</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="Deallocate">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>440</y>
           <width>151</width>
           <height>51</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Deallocate</string>
         </property>
        </widget>
       </widget>
       <widget class="QPushButton" name="Update">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>730</y>
          <width>221</width>
          <height>51</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Arial</family>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">#Update {
	color: white;
	background-color: #2779e3;
	border-width: 3px;
	border-radius: 5px
}</string>
        </property>
        <property name="text">
         <string>Update</string>
        </property>
       </widget>
       <widget class="QPushButton" name="Clear">
        <property name="geometry">
         <rect>
          <x>260</x>
          <y>730</y>
          <width>221</width>
          <height>51</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Arial</family>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">#Clear {
	color: white;
	background-color: rgb(255, 0, 0);
	border-width: 3px;
	border-radius: 5px;
}
</string>
        </property>
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
       <widget class="QWidget" name="InitWidget" native="true">
        <property name="geometry">
         <rect>
          <x>19</x>
          <y>120</y>
          <width>461</width>
          <height>151</height>
         </rect>
        </property>
        <widget class="QWidget" name="verticalLayoutWidget_2">
         <property name="geometry">
          <rect>
           <x>-1</x>
           <y>-1</y>
           <width>191</width>
           <height>151</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_1">
          <property name="leftMargin">
           <number>25</number>
          </property>
          <item>
           <widget class="QLabel" name="label_1">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Memory Size</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Holes Addresses</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Holes Sizes</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="verticalLayoutWidget_3">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>0</y>
           <width>271</width>
           <height>151</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="rightMargin">
           <number>25</number>
          </property>
          <item>
           <widget class="QLineEdit" name="MemSize">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="HolesAddresses">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="HolesSizes">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="AllocWidget" native="true">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>340</y>
          <width>461</width>
          <height>151</height>
         </rect>
        </property>
        <widget class="QWidget" name="verticalLayoutWidget_4">
         <property name="geometry">
          <rect>
           <x>-1</x>
           <y>-1</y>
           <width>191</width>
           <height>101</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="leftMargin">
           <number>25</number>
          </property>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Segments Names</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Segments Sizes</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="verticalLayoutWidget_5">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>0</y>
           <width>271</width>
           <height>101</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="rightMargin">
           <number>25</number>
          </property>
          <item>
           <widget class="QLineEdit" name="SegNames">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="SegSizes">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="horizontalLayoutWidget">
         <property name="geometry">
          <rect>
           <x>-1</x>
           <y>99</y>
           <width>461</width>
           <height>51</height>
          </rect>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="leftMargin">
           <number>25</number>
          </property>
          <property name="rightMargin">
           <number>25</number>
          </property>
          <item>
           <widget class="QRadioButton" name="BestFit">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Best-Fit</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="FirstFit">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>First-Fit</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="DeallocWidget" native="true">
        <property name="geometry">
         <rect>
          <x>19</x>
          <y>559</y>
          <width>461</width>
          <height>151</height>
         </rect>
        </property>
        <widget class="QWidget" name="verticalLayoutWidget_6">
         <property name="geometry">
          <rect>
           <x>-1</x>
           <y>-1</y>
           <width>461</width>
           <height>151</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="leftMargin">
           <number>25</number>
          </property>
          <property name="rightMargin">
           <number>25</number>
          </property>
          <item>
           <widget class="QRadioButton" name="PreAllocated">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Pre Allocated</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="UserAllocated">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>User Allocated</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="DeallocNum">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QLabel" name="label">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>10</y>
          <width>461</width>
          <height>51</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Arial</family>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Select an Operation</string>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="MemoryWidget" native="true">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>50</y>
      <width>451</width>
      <height>801</height>
     </rect>
    </property>
    <widget class="QGraphicsView" name="MemoryView">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>450</width>
       <height>800</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>450</width>
       <height>800</height>
      </size>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="SegmentsWidget" native="true">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>50</y>
      <width>451</width>
      <height>801</height>
     </rect>
    </property>
    <widget class="QGraphicsView" name="SegmentsView">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>450</width>
       <height>800</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>450</width>
       <height>800</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>Initialize</sender>
   <signal>clicked()</signal>
   <receiver>InitWidget</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Allocate</sender>
   <signal>clicked()</signal>
   <receiver>DeallocWidget</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>685</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Allocate</sender>
   <signal>clicked()</signal>
   <receiver>AllocWidget</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>466</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Allocate</sender>
   <signal>clicked()</signal>
   <receiver>InitWidget</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Initialize</sender>
   <signal>clicked()</signal>
   <receiver>DeallocWidget</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>685</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Initialize</sender>
   <signal>clicked()</signal>
   <receiver>AllocWidget</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>466</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Deallocate</sender>
   <signal>clicked()</signal>
   <receiver>DeallocWidget</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>585</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>685</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Deallocate</sender>
   <signal>clicked()</signal>
   <receiver>AllocWidget</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>585</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>466</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Deallocate</sender>
   <signal>clicked()</signal>
   <receiver>InitWidget</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>585</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
